### PR TITLE
Use the Azure.Core ETag type in the DPS service client

### PR DESCRIPTION
**Pull request by timtay-microsoft**
This is the first class we've added from the Azure.Core package to this library, so we now have a dependency on that package

### CODE REVIEW

File: iothub/device/src/Transport/Http/HttpClientHelper.cs [hunk 1]
```diff
@@ -99,19 +99,7 @@ private static async Task<T> ReadResponseMessageAsync<T>(HttpResponseMessage mes
                 return (T)(object)message;
             }
 
-            T entity = await ReadAsAsync<T>(message.Content, token).ConfigureAwait(false);
-
-            // ETag in the header is considered authoritative
-            if (entity is IETagHolder eTagHolder)
```
Comment (timtay-microsoft) [inline iothub/device/src/Transport/Http/HttpClientHelper.cs:105]: No class still implemented this interface in our IoT hub device client, so this if block was never entered.

Comment (timtay-microsoft) [reply iothub/device/src/Transport/Http/HttpClientHelper.cs:105]: It's worth remembering that HTTP never supported twin operations, and twin operations are the only ones that use ETags in the IoT hub device client

File: provisioning/service/src/Models/EnrollmentGroup.cs [hunk 2]
```diff
@@ -122,86 +123,10 @@ public EnrollmentGroup(string enrollmentGroupId, Attestation attestation)
...
-        /// <param name="initialTwinState">The <see cref="TwinState"/> with the initial Twin condition. This is optional and can be null.</param>
-        /// <param name="provisioningStatus">The <see cref="ProvisioningStatus"/> that determine the initial status of the device. This is optional and can be null.</param>
-        /// <param name="createdDateTimeUtc">The DateTime with the date and time that the enrollment was created. This is optional and can be null.</param>
-        /// <param name="lastUpdatedDateTimeUtc">The DateTime with the date and time that the enrollment was updated. This is optional and can be null.</param>
-        /// <param name="eTag">The string with the eTag that identify the correct instance of the enrollment in the service. It cannot be null or empty.</param>
-        /// <param name="capabilities">The capabilities of the device (ie: is it an edge device?)</param>
-        /// <exception cref="DeviceProvisioningServiceException">If the received JSON is invalid.</exception>
-        [JsonConstructor]
-        internal EnrollmentGroup(
```
Comment (timtay-microsoft) [inline provisioning/service/src/Models/EnrollmentGroup.cs:173]: We only need an empty ctor to make serialization work. I don't know why we had this uber ctor, but it isn't referenced anywhere so it seems safe to replace with an empty ctor

File: provisioning/service/src/DeviceRegistrationsClient.cs [hunk 3]
```diff
@@ -8,6 +8,7 @@
 using System.Threading.Tasks;
 using System.Threading;
 using Newtonsoft.Json;
+using Azure;
```
Comment (drwill-ms) [inline provisioning/service/src/DeviceRegistrationsClient.cs:11]: sort

File: provisioning/service/src/EnrollmentGroupsClient.cs [hunk 4]
```diff
@@ -9,6 +9,7 @@
 using System.Threading.Tasks;
 using System.Threading;
 using Newtonsoft.Json;
+using Azure;
```
Comment (drwill-ms) [inline provisioning/service/src/EnrollmentGroupsClient.cs:12]: sort

File: provisioning/service/src/Http/ContractApiHttp.cs [hunk 5]
```diff
@@ -127,7 +128,12 @@ internal class ContractApiHttp : IContractApiHttp
                         msg.Headers.Add(header.Key, header.Value);
                     }
                 }
-                InsertIfMatch(msg, ifMatch);
+                
+                if (!string.IsNullOrWhiteSpace(eTag.ToString()))
```
Comment (drwill-ms) [inline provisioning/service/src/Http/ContractApiHttp.cs:132]: Can ETag be null?

File: provisioning/service/src/IndividualEnrollmentsClient.cs [hunk 6]
```diff
@@ -9,6 +9,7 @@
 using System.Threading.Tasks;
 using System.Threading;
 using Newtonsoft.Json;
+using Azure;
```
Comment (drwill-ms) [inline provisioning/service/src/IndividualEnrollmentsClient.cs:12]: sort

**Review by drwill-ms — APPROVED** [review]

File: provisioning/service/src/Http/ContractApiHttp.cs (hunk 5 above)
Comment (timtay-microsoft) [reply provisioning/service/src/Http/ContractApiHttp.cs:132]: No

**Pull request merged**